fv3core/stencils/delnflux.py

```diff
@@ -1,218 +1,627 @@
+    fy: FloatField,
 ):
     with computation(PARALLEL), interval(...):
-        fy2[0, 0, 0] = (
+        fy[0, 0, 0] = (
             0.5 * (sin_sg4[0, -1, 0] + sin_sg2) * dx * (q[0, -1, 0] - q) * rdyc
         )
 
 
-@gtstencil
-def d2_highorder(fx2: FloatField, fy2: FloatField, rarea: FloatFieldIJ, d2: FloatField):
-    with computation(PARALLEL), interval(...):
-        d2[0, 0, 0] = (fx2 - fx2[1, 0, 0] + fy2 - fy2[0, 1, 0]) * rarea
-
-
-@gtstencil
-def d2_damp(q: FloatField, d2: FloatField, damp: float):
-    with computation(PARALLEL), interval(...):
-        d2[0, 0, 0] = damp * q
+def d2_highorder_stencil(
+    fx: FloatField, fy: FloatField, rarea: FloatFieldIJ, d2: FloatField
+):
+    from __externals__ import nord0, nord1, nord2, nord3
+
+    with computation(PARALLEL), interval(0, 1):
+        if __INLINED(nord0 > 0):
+            d2 = d2_highorder(fx, fy, rarea)
+    with computation(PARALLEL), interval(1, 2):
+        if __INLINED(nord1 > 0):
+            d2 = d2_highorder(fx, fy, rarea)
+    with computation(PARALLEL), interval(2, 3):
+        if __INLINED(nord2 > 0):
+            d2 = d2_highorder(fx, fy, rarea)
+    with computation(PARALLEL), interval(3, None):
+        if __INLINED(nord3 > 0):
+            d2 = d2_highorder(fx, fy, rarea)
+
+
+@gtscript.function
+def d2_highorder(fx: FloatField, fy: FloatField, rarea: FloatField):
+    d2 = (fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) * rarea
+    return d2
+
+
+def d2_damp_interval(q: FloatField, d2: FloatField, damp: FloatFieldK):
+    from __externals__ import (
+        local_ie,
+        local_is,
+        local_je,
+        local_js,
+        nord0,
+        nord1,
+        nord2,
+        nord3,
+    )
 
+    with computation(PARALLEL), interval(0, 1):
+        if __INLINED(nord0 == 0):
+            with horizontal(
+                region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+            ):
+                d2[0, 0, 0] = damp * q
+        else:
+            d2[0, 0, 0] = damp * q
+    with computation(PARALLEL), interval(1, 2):
+        if __INLINED(nord1 == 0):
+            with horizontal(
+                region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+            ):
+                d2[0, 0, 0] = damp * q
+        else:
+            d2[0, 0, 0] = damp * q
+    with computation(PARALLEL), interval(2, 3):
+        if __INLINED(nord2 == 0):
+            with horizontal(
+                region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+            ):
+                d2[0, 0, 0] = damp * q
+        else:
+            d2[0, 0, 0] = damp * q
+    with computation(PARALLEL), interval(3, None):
+        if __INLINED(nord3 == 0):
+            with horizontal(
+                region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+            ):
+                d2[0, 0, 0] = damp * q
+        else:
+            d2[0, 0, 0] = damp * q
+
+
+def copy_stencil_interval(q_in: FloatField, q_out: FloatField):
```

Comment:
I am fine with this just being here as is is only used here, but was there thought of putting it in `basic_operations`?